pl_bolts/models/rl/ppo_model.py

```diff
@@ -0,0 +1,373 @@
+
+import pytorch_lightning as pl
+import torch
+from torch.optim.optimizer import Optimizer
+from torch.utils.data import DataLoader
+
+from pl_bolts.datamodules import ExperienceSourceDataset
+from pl_bolts.models.rl.common.networks import ActorCategorical, ActorContinous, MLP
+from pl_bolts.utils import _GYM_AVAILABLE
+from pl_bolts.utils.warnings import warn_missing_pkg
+
+if _GYM_AVAILABLE:
+    import gym
+else:  # pragma: no cover
+    warn_missing_pkg('gym')
+
+
+class PPO(pl.LightningModule):
+    """
+    PyTorch Lightning implementation of Proximal Policy Optimization
+    <https://arxiv.org/abs/1707.06347>`_
+    Paper authors: John Schulman, Filip Wolski, Prafulla Dhariwal, Alec Radford, Oleg Klimov
+
+    Model implemented by:
+    `Sidhant Sundrani <https://github.com/sid-sundrani>`
+
+    Example:
+        >>> from pl_bolts.models.rl.ppo_model import PPO
+        ...
+        >>> model = PPO("CartPole-v0")
+
+    Note:
+        This example is based on:
+        https://github.com/openai/baselines/blob/master/baselines/ppo2/ppo2.py
+        https://github.com/openai/spinningup/blob/master/spinup/algos/pytorch/ppo/ppo.py
+
+    Note:
+        Currently only supports CPU and single GPU training with `distributed_backend=dp`
+    """
+
+    def __init__(
+        self,
+        env: str,
+        gamma: float = 0.99,
+        lam: float = 0.95,
+        lr_actor: float = 3e-4,
+        lr_critic: float = 1e-3,
+        max_episode_len: float = 200,
+        batch_size: int = 512,
+        steps_per_epoch: int = 2048,
+        nb_optim_iters: int = 4,
+        clip_ratio: float = 0.2,
+        **kwargs,
+    ) -> None:
+        """
+        Args:
+            env: gym environment tag
+            gamma: discount factor
+            lam: advantage discount factor (lambda in the paper)
+            lr_actor: learning rate of actor network
+            lr_critic: learning rate of critic network
+            max_episode_len: maximum number interactions (actions) in an episode
+            batch_size:  batch_size when training network- can simulate number of policy updates performed per epoch
+            steps_per_epoch: how many action-state pairs to rollout for trajectory collection per epoch
+            nb_optim_iters: how many steps of gradient descent to perform on each batch
+            clip_ratio: hyperparameter for clipping in the policy objective
+        """
+        super().__init__()
+
+        if not _GYM_AVAILABLE:
+            raise ModuleNotFoundError('This Module requires gym environment which is not installed yet.')
+
+        # Hyperparameters
+        self.lr_actor = lr_actor
+        self.lr_critic = lr_critic
+        self.steps_per_epoch = steps_per_epoch
+        self.nb_optim_iters = nb_optim_iters
+        self.batch_size = batch_size
+        self.gamma = gamma
+        self.lam = lam
+        self.max_episode_len = max_episode_len
+        self.clip_ratio = clip_ratio
+        self.save_hyperparameters()
```

Comment:
How about saving these hyperparameters with:
```suggestion
        # Hyperparameters
        self.save_hyperparameters()
```
and referencing them with `self.hparams.lr_actor`?

Reply:
I'm thinking we can keep it as is for now because then it will be consistent with the other RL implementations, such as reinforce_model.py, dqn_model and vpg. we can refactor it all at once later for consistency. thoughts?